### ISSUE TITLE
`compression_wrapper` compatibility issue

### ISSUE THREAD

**Issue by piskvorky** [RaRe-Technologies/smart_open#332]
This used to work in 1.5.6, perhaps even later:

```python
import smart_open

some_fp = io.BytesIO()  # any file-like object
…
wrapped = smart_open.compression_wrapper(some_fp, some_filename, 'rb')
```

It doesn't work now, `smart_open.compression_wrapper` no longer exists.

What's the reason? I cannot find anything in the CHANGELOG.

**Comment by piskvorky** [comment]
Upon investigation, it seems like:

```python
from smart_open.smart_open_lib import _compression_wrapper
smart_open.compression_wrapper = _compression_wrapper
```

will restore this (desirable) functionality back.

@mpenkov I'd strongly prefer to avoid such shenanigans. Do you see a better way?

If not, can we bring back `compression_wrapper` (and perhaps others who suffered a similar fate)?

**Comment by mpenkov** [comment]
What is the use case here? What are you attempting to achieve, and what are the parameters you're passing to the wrapper?

**Comment by piskvorky** [comment]
Parameter 1 = `io.BufferedIOBase`, parameter 2 = filename `str`, parameter 3 = mode = 'rb'.

We're wrapping an open file-like object so that it decompresses transparently.

**Comment by mpenkov** [comment]
I demoted compression_wrapper to be an internal function (it's called _compression_wrapper) when I redesigned the smart_open library [over a year ago](https://github.com/RaRe-Technologies/smart_open/commit/b499f874f91df67761b0c90e0b2b894f1f908cc3). My reasoning for making it internal was:

- The function wasn't publicly documented anywhere, unlike e.g. the obviously public smart_open and s3_iter
- There was no example code for using that function
- There were no unit tests exercising that function on its own

The main benefits of demoting such functions to internal is a cleaner API.

Here's a better way to satisfy your use case than the import-rename pattern you suggested:

```python
>>> import io, gzip
>>>
>>> # prepare some gzipped data
>>> buf = io.BytesIO()
>>> with gzip.GzipFile(fileobj=buf, 'w') as fout: fout.write(b'this is a bytestring')
>>> buf.seek(0)
0
>>> # use case starts here
>>> buf.name = 'file.gz' # add a .name attribute so smart_open knows what compressor to use
>>> import smart_open
>>> smart_open.open(buf, 'rb').read() 
b'this is a bytestring'
```

If your file object already has an appropriate .name, then you don't have to add one.
